### PR TITLE
Introduce `open` method no Debt Order

### PR DESCRIPTION
**Pull request by saturnial**

### CODE REVIEW

File: src/debt_order/index.ts [hunk 1]
```diff
@@ -1,19 +1,47 @@
...
 import { Dharma } from "../";
+import { CollateralizedSimpleInterestLoanOrder } from "../adapters/collateralized_simple_interest_loan_adapter";
 import { DebtOrderData, ECDSASignature, InterestRate, Term, TokenAmount } from "../types";
 
+export interface DebtOrderParams {
+    principal: TokenAmount;
+    collateral: TokenAmount;
+    interestRate: InterestRate;
+    term: Term;
+    debtor: string;
```
Comment (chrismin) [inline src/debt_order/index.ts]: Can we standardize this to either `debtor` or `debtorAddress`?  For the creditor side, I've used `creditorAddress`.  I feel the latter is more specific but am not married to either.

**Comment by saturnial** [comment]
♻️ @chrismin @graemecode

File: src/debt_order/index.ts [hunk 2]
```diff
@@ -14,16 +21,33 @@ import { BLOCK_TIME_ESTIMATE_SECONDS } from "../../utils/constants";
...
+    public async open(params: DebtOrderParams) {
+        const { principal, collateral, interestRate, term, debtorAddress } = params;
+
+        const loanOrder: CollateralizedSimpleInterestLoanOrder = {
+            principalAmount: principal.rawAmount,
+            principalTokenSymbol: principal.tokenSymbol,
+
+            interestRate: interestRate.raw,
+            amortizationUnit: term.unit,
+            termLength: term.length(),
```
Comment (graemecode) [inline src/debt_order/index.ts]: I must have missed this in the discussion around `term`, but I think this should be `getLength()` -- length() is a little confusing to me.

Comment (saturnial) [reply src/debt_order/index.ts]: 💯 I think we should also do a rev on these intermediary types to ensure we have nice APIs here as well. I'm just ripping on them without much thought.

**Pull request merged**